### ISSUE TITLE
post/contentNode types return invalid databaseId when querying for draft post

### ISSUE THREAD

**Issue by blakewilson** [wp-graphql/wp-graphql#2832]
### Description

When querying for a draft post by `databaseId` to the `post`/`contentNode` types, the returned post/contentNode has an incorrect `databaseId`

### Steps to reproduce

1. Create a post with a title and content and save it as a draft. Make a note of the ID that is generated for the draft
2. Make the following query in WPGraphQL IDE:
   ```graphql
    query GetDraftPost($id: ID!) {
      contentNode(id: $id, idType: DATABASE_ID, asPreview:true) {
        databaseId
        slug
        ... on NodeWithTitle {
          title
        }
        ... on NodeWithContentEditor {
          content
        }
      }
    }
   ```
3. Set the query variables to a JSON object with an `id` property with the value of the draft post database ID.
4. Make the request, and notice the contentNode comes back fine with the proper title and content but the database ID is different (usually it has been incremented by 1)

### WPGraphQL Version

1.14.3

### WordPress Version

6.2.2

### PHP Version

7.4.30

### Please confirm that you have searched existing issues in the repo.

- [X] Yes

### Please confirm that you have disabled ALL plugins except for WPGraphQL.

- [X] Yes
- [ ] My issue is with compatibility with a specific WordPress plugin, and I have listed all my installed plugins (and version info) above.

**Comment by jasonbahl** [comment]
@blakewilson this is expected behavior. 

Previews are revisions and have different ids than the thing being previewed. 

When making changes, the changes are stored on another post (revision) not the post itself. 

So a preview should be expected to have a different databaseId as it's a different node

**Comment by justlevine** [comment]
Hey @blakewilson pretty sure that's expected behavior. The post ID should be available as `parentDatabaseID` (see here: https://github.com/wp-graphql/wp-graphql/blob/0f330b681493774b3242718aa43d1e3e8487a690/src/Model/Post.php#L128)

**Comment by blakewilson** [comment]
Makes sense. To make things worse I kept getting null back from the `521` databaseId when I was expected the preview... but I wasn't authenticated RIP. Thanks for clarifying!
